lib/core/sdam/server.js

```diff
@@ -249,6 +240,11 @@ class Server extends EventEmitter {
       (callback = options), (options = {}), (options = options || {});
     }
 
+    if (this.s.state === STATE_CLOSING || this.s.state === STATE_CLOSED) {
+      callback(new MongoError('server is closed'));
```

Comment:
Unfortunate that this needs to be duplicated over and over.

Reply:
Yeah this is rough today, but will also hopefully go away with CMAP - you won't be able to even get to one of these wire protocol methods if there is no connection available.